aws/resource_aws_glue_resource_policy.go

```diff
@@ -0,0 +1,85 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsGlueResourcePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueResourcePolicyCreate,
+		Read:   resourceAwsGlueResourcePolicyRead,
+		Update: resourceAwsGlueResourcePolicyUpdate,
+		Delete: resourceAwsGlueResourcePolicyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"policy": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueResourcePolicyCreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
at a closer look, I think we can compress the create/update functions to a single `resourceAwsGlueResourcePolicyPut` see `resourceAwsApiGatewayGatewayResponse` resource for an example

Reply:
There is a small difference the `PolicyExistsCondition` condition changes but I wrapped it in a function that now looks something like this: 
`Create: resourceAwsGlueResourcePolicyPut(glue.ExistConditionNotExist),`